dist/stat_data.py

```diff
@@ -397,6 +397,27 @@ def __init__(self, name, desc, flags=''):
     HistoryStat('hs_gc_pages_evict', 'history pages added for eviction during garbage collection'),
     HistoryStat('hs_gc_pages_removed', 'history pages removed for garbage collection'),
     HistoryStat('hs_gc_pages_visited', 'history pages visited for garbage collection'),
+    HistoryStat('hs_write_addr_prepared', 'page addresses written that included at least one prepare state'),
+    HistoryStat('hs_write_addr_newest_start_durable_ts', 'page addresses written that include a newest start durable timestamp '),
+    HistoryStat('hs_write_addr_newest_stop_durable_ts', 'page addresses written that include a newest stop durable timestamp '),
+    HistoryStat('hs_write_addr_oldest_start_ts', 'page addresses written that include a oldest start timestamp '),
+    HistoryStat('hs_write_addr_oldest_start_txn', 'page addresses written that include a oldest start transaction ID '),
+    HistoryStat('hs_write_addr_newest_stop_ts', 'page addresses written that include a newest stop timestamp '),
+    HistoryStat('hs_write_addr_newest_stop_txn', 'page addresses written that include a newest stop transaction ID'),
+    HistoryStat('hs_write_pages_prepared', 'pages written that included at least one prepare state'),
+    HistoryStat('hs_write_pages_durable_start_ts', 'pages written that included at least one start durable timestamp'),
+    HistoryStat('hs_write_pages_durable_stop_ts', 'pages written that included at least one stop durable timestamp'),
+    HistoryStat('hs_write_pages_start_ts', 'pages written that included at least one start timestamp'),
+    HistoryStat('hs_write_pages_start_txn', 'pages written that included at least one start transaction ID'),
+    HistoryStat('hs_write_pages_stop_ts', 'pages written that included at least one stop timestamp'),
+    HistoryStat('hs_write_pages_stop_txn', 'pages written that included at least one stop transaction ID'),
+    HistoryStat('hs_write_rec_prepared', 'records written that included a prepare state'),
+    HistoryStat('hs_write_rec_durable_start_ts', 'records written that included a start durable timestamp'),
+    HistoryStat('hs_write_rec_durable_stop_ts', 'records written that included a stop durable timestamp'),
+    HistoryStat('hs_write_rec_start_ts', 'records written that included a start timestamp'),
+    HistoryStat('hs_write_rec_start_txn', 'records written that included a start transaction ID'),
+    HistoryStat('hs_write_rec_stop_ts', 'records written that included a stop timestamp'),
+    HistoryStat('hs_write_rec_stop_txn', 'records written that included a stop transaction ID'),
```

Comment:
Is it interesting to tease apart start txn IDs vs. start timestamps? In other code I've not done that, I've just say "a start thing got set", so there are 3 categories, "start", "stop" and "durable". If it's not interesting to split them apart, that simplifies things.

Reply:
I believe it is interesting to have different stats for txnID vs timestamp. Generally I expect that the window of required timestamps can be long seeing issues because of that might point towards suggesting applications change how they use MongoDB.
I'm hoping we pretty much never see transaction IDs written to disk, if we do that's a pointer to somewhere we should look to make a WiredTiger of MongoDB change to avoid.